herbert_core/utilities/classes/@object_lookup/object_lookup.m

```diff
@@ -149,484 +182,374 @@
-            %   but the same size as the object
-            % - If the object is non-empty array, returns a scalar structure corresponding
-            %   to the the first element in the array of objects
+
+        function obj = check_combo_arg (obj)
+            % Verify interdependent variables and the validity of the
+            % obtained serializable object. Return the result of the check.
             %
+            % Recompute any cached arguments.
             %
-            % See also structPublic, structArrIndep, structArrPublic
-            
-            names = obj.propNamesIndep_';
-            if ~isempty(obj)
-                tmp = obj(1);
-                S = obj.scalarEmptyStructIndep_;
-                for i=1:numel(names)
-                    S.(names{i}) = tmp.(names{i});
-                end
-            else
-                args = [names; repmat({cell(size(obj))},size(names))];
-                S = struct(args{:});
+            % Throw an error if the properties are inconsistent and return
+            % without problem it they are not.
+
+            % Check the number of index arrays and the number of size vectors are
+            % the same. This gives the number of stored arrays
+            if numel(obj.indx_) ~= numel(obj.sz_)
+                error('HERBERT:object_lookup:invalid_argument', ...
+                    ['The number of stored array sizes is inconsistent with the ',...
+                    'number of index arrays'])
             end
-        end
-        
-        function S = structArrIndep(obj)
-            % Return the independent properties of an object array as a structure array
-            %
-            %   >> s = structArrIndep(obj)
-            %
-            % Use <a href="matlab:help('structIndep');">structIndep</a> for behaviour that more closely matches the Matlab
-            % intrinsic function struct.
-            %
-            % Has the same behaviour as the Matlab instrinsic struct in that:
-            % - Any structure array is returned unchanged
-            % - If an object is empty, an empty structure is returned with fieldnames
-            %   but the same size as the object
-            %
-            % However, differs in the behaviour if an object array:
-            % - If the object is non-empty array, returns a structure array of the same
-            %   size. This is different to the instrinsic Matlab, which returns a scalar
-            %   structure from the first element in the array of objects
-            %
-            %
-            % See also structIndep, structPublic, structArrPublic
             
-            if numel(obj)>1
-                S = arrayfun(@fill_it, obj);
-            else
-                S = structIndep(obj);
+            % Check the sizes of the stored object arrays are consistent with the
+            % length of the index arrays
+            nel = cellfun(@prod, obj.sz_);
```

Comment:
```suggestion
            nel = obj.nelmts;
```

Reply:
I've generally preferred to use the private properties directly rather than the public getters on efficiency grounds, but this is actually something I want to talk with you about on general design principles and private .v. public properties coming out of thinking about the design of serializable. I'm leaving as is for the time-being until I have a better mental model.